languages/julia/config.json

```diff
@@ -9,6 +9,14 @@
     },
     "exercises": {
         "concept": [
+            {
+                "uuid": "39ebdd04-4f84-4817-bf9d-f1f9e066c283",
+                "slug": "lasagna",
+                "concepts": [
+                    "methods.basic"
```

Comment:
This goes against our recently suggested [naming policy](https://github.com/exercism/v3/blob/master/docs/maintainers/determining-concepts.md#naming-concepts). Was that intentional? :)

Reply:
Yes. I've explained it on [Slack](https://exercism-team.slack.com/archives/CR91YFNG3/p1585922728283300), should've commented here, too, sorry!

The concepts used here are to be defined in a file `methods.md` (which hasn't been written yet fwiw). That file is also going to cover things like named arguments and optional arguments. Splitting it up into various files `methods-positional-args.md` would come with an immense overhead while also making it harder to read the full concept docs since they'd either have some kind of reading order/hierarchy or lots of redundant parts. Having a concept `methods.basic` allows me to say "the concept introduced in this exercise is defined in `methods.md` to other maintainers/contributors" without requiring an additional file.
The actual progression logic in `config.json` won't have to care about it and can just treat it like any other concept, but it allows me to verify that the concept is well-defined for all contributors via an automated check.

If you have a better suggestion for how I can ensure that the _concept in config.json is defined somewhere_ without this naming scheme, I'm more than happy to drop the dots.

---

cc @iHiD: Here's a real world example :)

Reply:
> The actual progression logic in config.json won't have to care about it and can just treat it like any other concept,

The problem is students will see "methods.basics" as a concept they've learnt in the website, so I think it does matter that these are named to be human friendly.

---

> If you have a better suggestion for how I can ensure that the concept in config.json is defined somewhere without this naming scheme, I'm more than happy to drop the dots.

I think the key area where out thoughts diverge on this is that for me the concept docs are a transitional thing to help us get going with v3. By the time all the tracks are exploded back out to their own repos, I imagine the concept docs will be a reference but nothing more. I don't see them as some sort of canonical set of named elements - just a folder of reference files. Am I right in thinking that you see them as much more first-class citizens of v3?

Reply:
> The problem is students will see "methods.basics" as a concept they've learnt in the website, so I think it does matter that these are named to be human friendly.

I wasn't aware that these are student-facing. But I don't see how methods.basic is any less human-friendly than methods-basic. Granted, `basic` isn't all that descriptive but that doesn't change the need for an annotation of some sort. For all I care we can turn them into dashes in the student-facing UI, too.

> I think the key area where out thoughts diverge on this is that for me the concept docs are a transitional thing to help us get going with v3. By the time all the tracks are exploded back out to their own repos, I imagine the concept docs will be a reference but nothing more. [...] Am I right in thinking that you see them as much more first-class citizens of v3?

No, I also see them as a transitional, or at least contributor-facing, thing.

I see them as the definition of each concept to avoid situations in which contributors have different definitions in mind. One example related to this PR would be the difference between functions and methods. There's a somewhat common misunderstanding among Julia users about that difference, particularly among people who have a background in class-based object-oriented languages. I think it's quite important to have a common definition of each concept, when developing and implementing the exercises. Note that the content of them is irrelevant in this context. They could contain nothing more than a link to the manual for example.

I think this isn't far off to your view of them as a reference, except that I want to enforce and automatically check that they exist. That last part is the sole reason why I want to use the dot syntax because it makes handling the concepts more convenient.

Reply:
OK - I understand more now. Thank you.

> I wasn't aware that these are student-facing. But I don't see how methods.basic is any less human-friendly than methods-basic. 

So in our new naming-schema methods-basic would be just as invalid. Neither are proper descriptive names - they're both computer-facing really and `methods-basic` is **more** confusing. I would have it as `introduction-to-methods` or something similar. 

I feel like I'd prefer an `introduction-to-methods` concept, and then have something somewhere (track's `config.json`?) that maps concepts to files for the purpose you suggest. (I think someone suggested making redirect files - this would sort of be like this but without the extra md files)

I really feel we're letting our CI drive out implementation otherwise.

Reply:
> I feel like I'd prefer an introduction-to-methods concept, and then have something somewhere (track's config.json?) that maps concepts to files for the purpose you suggest.
>
> I really feel we're letting our CI drive out implementation otherwise.

We could map it in `concepts.csv` but it's less convenient than having it in the name. I didn't know that these were student-facing at all. With that in mind, I can understand your concerns a lot better now, and they justify the added inconvenience :)

---

> So in our new naming-schema methods-basic would be just as invalid. Neither are proper descriptive names - they're both computer-facing really and methods-basic is more confusing. I would have it as introduction-to-methods or something similar.

This goes beyond the initial question here, but I'd consider `introduction-to-methods` quite a terrible name for keeping track of the progression in `config.json`. How about adding a "pretty name" for the purpose of displaying it on the website?

Reply:
@iHiD should we move this to an issue?